Quoting 'info '(automake)Extending aclocal'':

Starting with Automake 1.8, `aclocal' will warn about all
underquoted calls to `AC_DEFUN'.  We realize this will annoy a lot of
people, because `aclocal' was not so strict in the past and many third
party macros are underquoted; and we have to apologize for this
temporary inconvenience.

--- apt-0.5.15cnc6.orig/buildlib/tools.m4	2003-03-26 10:45:36 +1200
+++ apt-0.5.15cnc6/buildlib/tools.m4	2004-05-14 22:00:16 +1300
@@ -1,4 +1,4 @@
-AC_DEFUN(ah_HAVE_GETCONF,
+AC_DEFUN([ah_HAVE_GETCONF],
 	[AC_ARG_WITH(getconf,
 		[  --with-getconf          Enable automagical buildtime configuration],
 		[if test "$withval" = "yes"; then
@@ -14,7 +14,7 @@ AC_DEFUN(ah_HAVE_GETCONF,
 ])
 
 dnl ah_GET_CONF(variable, value ..., [default])
-AC_DEFUN(ah_GET_GETCONF,
+AC_DEFUN([ah_GET_GETCONF],
 	[AC_REQUIRE([ah_HAVE_GETCONF])
 	if test ! -z "$GETCONF";then
 		old_args="[$]@"
@@ -28,7 +28,7 @@ AC_DEFUN(ah_GET_GETCONF,
 		eval $1="$3"
 	fi
 ])
-AC_DEFUN(ah_NUM_CPUS,
+AC_DEFUN([ah_NUM_CPUS],
 	[AC_MSG_CHECKING([number of cpus])
 	AC_ARG_WITH(cpus,
 		[  --with-cpus             The number of cpus to be used for building(see --with-procs, default 1)],
@@ -56,7 +56,7 @@ AC_DEFUN(ah_NUM_CPUS,
 	AC_MSG_RESULT([$ah_NUM_CPUS_msg])
 	AC_SUBST(NUM_CPUS)
 ])
-AC_DEFUN(ah_PROC_MULTIPLY,
+AC_DEFUN([ah_PROC_MULTIPLY],
 	[AC_REQUIRE([ah_NUM_CPUS])
 	AC_MSG_CHECKING([processor multiplier])
 	AC_ARG_WITH(proc-multiply,
@@ -72,7 +72,7 @@ AC_DEFUN(ah_PROC_MULTIPLY,
 	AC_SUBST(PROC_MULTIPLY)
 ])
 
-AC_DEFUN(ah_NUM_PROCS,
+AC_DEFUN([ah_NUM_PROCS],
 	[AC_REQUIRE([ah_PROC_MULTIPLY])
 	AC_REQUIRE([ah_NUM_CPUS])
 	AC_MSG_CHECKING([number of processes to run during make])
@@ -89,7 +89,7 @@ AC_DEFUN(ah_NUM_PROCS,
 	AC_SUBST(NUM_PROCS)
 ])
 
-AC_DEFUN(rc_GLIBC_VER,
+AC_DEFUN([rc_GLIBC_VER],
 	[AC_MSG_CHECKING([glibc version])
 	dummy=if$$
 	cat <<_GLIBC_>$dummy.c
@@ -111,7 +111,7 @@ _GLIBC_
 	AC_SUBST(GLIBC_VER)
 ])
 
-AC_DEFUN(rc_LIBSTDCPP_VER,
+AC_DEFUN([rc_LIBSTDCPP_VER],
 	[AC_MSG_CHECKING([libstdc++ version])
 	dummy=if$$
 	cat <<_LIBSTDCPP_>$dummy.cc
@@ -138,7 +138,7 @@ _LIBSTDCPP_
 	AC_SUBST(LIBSTDCPP_VER)
 ])
 
-AC_DEFUN(ah_GCC3DEP,[
+AC_DEFUN([ah_GCC3DEP],[
 	AC_MSG_CHECKING(if $CXX -MD works)
 	touch gcc3dep.cc
 	${CXX-c++} -MD -o gcc3dep_test.o -c gcc3dep.cc
